File: algoritmos-y-estructuras-de-datos-2/algorithms/sorting/selectionSort.js
function selectionSort(A) {
    for (let i = 0; i < A.length-1; i++) {
        let minIndex = i; 
        
        for (let j = i+1; j < A.length; j++) {
            if (A[j] < A[minIndex]) {
                minIndex = j
            }   
        }
        
        let aux = A[i];
        A[i] = A[minIndex];
        A[minIndex] = aux;
    }
    return A;
}


let A = [2, 0, 2, 1, 3, 5];

console.log(selectionSort(A));

File: algoritmos-y-estructuras-de-datos-2/algorithms/sorting/selectionSort.hs
main :: IO ()
main = do
  print (selectionSort example)


isSorted :: Ord a => [a] -> Bool
isSorted []  = True
isSorted [x] = True
isSorted (x:y:ys)
  | x <= y    = isSorted (y:ys)
  | otherwise = False
  
removeFirst :: Eq a => a -> [a] -> [a]
removeFirst _ [] = []
removeFirst x (y:ys)
  | x == y    = ys
  | otherwise = y : (removeFirst x ys)

selectionSort :: Ord a => [a] -> [a]
selectionSort []     = []
selectionSort [x]    = [x]
selectionSort (x:xs) = m : selectionSort (removeFirst m (x:xs))
  where m = minimum (x:xs)
  

example = [1, 0, 2, 24, -1]
